tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/CallByNameWrapHandler.scala

```diff
@@ -10,20 +10,29 @@ import at.forsyte.apalache.tla.lir.transformations.{TlaExTransformation, Transfo
  * no longer necessary to guard against preprocessing.
  */
 class CallByNameWrapHandler(tracker: TransformationTracker) {
-//  We create class wrappers, so `tr.getClass.getSimpleName` works in the passes
+  //  We create class wrappers, so `tr.getClass.getSimpleName` works in the passes
   sealed class Wrap(tr: TlaExTransformation) extends TlaExTransformation {
     override def apply(arg: TlaEx): TlaEx = tr(arg)
   }
+
   sealed class Unwrap(tr: TlaExTransformation) extends TlaExTransformation {
     override def apply(arg: TlaEx): TlaEx = tr(arg)
   }
 
   private def wrapEx(nameEx: NameEx) = OperEx(ApalacheOper.callByName, nameEx)(nameEx.typeTag)
 
+  // should we wrap an operator
+  private def isFold(op: TlaOper) = {
+    op == ApalacheOper.foldSet || op == ApalacheOper.foldSeq
+  }
```

Comment:
If you had made `TlaExUtil::isFold` public, you could have called it here. Still unnecessary with builtin `|`, either way.